source/containers.tex

```diff
@@ -378,97 +534,143 @@
- \tcode{reverse_iterator<iterator>}   &
- compile time                           \\ \rowsep
-\tcode{X::const_reverse_iterator}         &
- constant iterator type whose value type is \tcode{T}  &
- \tcode{reverse_iterator<const_iterator>}    &
- compile time                               \\ \rowsep
-\tcode{a.rbegin()}                  &
- \tcode{reverse_iterator; const_reverse_iterator} for constant \tcode{a} &
- \tcode{reverse_iterator(end())}   &
- constant                           \\ \rowsep
-\tcode{a.rend()}                     &
- \tcode{reverse_iterator; const_reverse_iterator} for constant \tcode{a} &
- \tcode{reverse_iterator(begin())}    &
- constant                           \\ \rowsep
-\tcode{a.crbegin()}         &
- \tcode{const_reverse_iterator}     &
- \tcode{const_cast<X const\&>(a).rbegin()} &
- constant                   \\ \rowsep
-\tcode{a.crend()}         &
- \tcode{const_reverse_iterator}     &
- \tcode{const_cast<X const\&>(a).rend()} &
- constant                   \\
-\end{libreqtab4a}
+\rSec3[container.rev.reqmts]{Reversible container requirements}
 
 \pnum
-Unless otherwise specified (see~\ref{associative.reqmts.except}, \ref{unord.req.except}, \ref{deque.modifiers}, and
-\ref{vector.modifiers})
-all container types defined in this Clause meet
-the following additional requirements:
-
+A type \tcode{X} meets the \defnadj{reversible}{container} requirements if
+\tcode{X} meets the container requirements,
+the iterator type of \tcode{X} belongs to the
+bidirectional or random access iterator categories\iref{iterator.requirements},
+and
+the following types and expressions are well-formed and have
+the specified semantics.
 \begin{itemize}
 \item
-if an exception is thrown by an
-\tcode{insert()} or \tcode{emplace()}
-function while inserting a single element, that
-function has no effects.
-\item
-if an exception is thrown by a
-\tcode{push_back()},
-\tcode{push_front()},
-\tcode{emplace_back()}, or \tcode{emplace_front()}
-function, that function has no effects.
-\item
-no
-\tcode{erase()},
-\tcode{clear()},
-\tcode{pop_back()}
-or
-\tcode{pop_front()}
-function throws an exception.
-\item
-no copy constructor or assignment operator of a returned iterator
-throws an exception.
-\item
-no
-\tcode{swap()}
-function throws an exception.
+The name \tcode{X::reverse_iterator} denotes
+the type \tcode{reverse_iterator<X::iterator>},
+an iterator type whose value type is \tcode{T}.
 \item
-no
-\tcode{swap()}
-function invalidates any references,
-pointers, or iterators referring to the elements
-of the containers being swapped.
-\begin{note}
-The \tcode{end()} iterator does not refer to any element, so it can be invalidated.
-\end{note}
+The name \tcode{X::const_reverse_iterator} denotes
+the type \tcode{reverse_iterator<X::const_iterator>},
+a constant iterator type whose value type is \tcode{T}.
+\end{itemize}
+
+\begin{itemexpr}
+a.rbegin()
+\end{itemexpr}
+
+\begin{itemdescr}
+\pnum
+\result
+\tcode{reverse_iterator};
```

Comment:
Shouldn't the _Result:_\s of the reversible container requirements also specify that they are prvalues?

Reply:
fixed.